packages/dashboard/src/components/iot-dashboard/iot-dashboard.tsx

```diff
@@ -1,475 +1,228 @@
+  onEditCellSizeAction,
+  onEditWidthAction,
+  onSelectAction,
+  SelectActionInput,
+} from '../../dashboard-actions/actions';
+import { DashboardReducerState, UndoQueue, DashboardStore, DashboardConfiguration } from '../../types';
+import { dashboardReducer } from '../../dashboard-actions/dashboardReducer';
+import { getRandomWidget } from '../../dashboard-actions/createWidget';
 import { trimWidgetPosition } from './trimWidgetPosition';
-import { deleteWidgets } from '../../dashboard-actions/delete';
-import { paste } from '../../dashboard-actions/paste';
+import { dashboardConfig } from '../../testing/mocks';
 
-const DEFAULT_STRETCH_TO_FIT = true;
-const DEFAULT_CELL_SIZE = 15;
+const DEFAULT_STRETCH_TO_FIT = false;
+
+const DEFAULT_CELL_SIZE = 10;
+const DEFAULT_WIDTH = 1000;
 
 @Component({
   tag: 'iot-dashboard',
-  styleUrl: 'iot-dashboard.css',
   shadow: false,
 })
 export class IotDashboard {
-  private resizer: ResizeObserver;
-
   /** The configurations which determines which widgets render where with what settings. */
   @Prop() dashboardConfiguration: DashboardConfiguration;
 
-  /**
-   * Callback that is fired every time the dashboard configuration has been altered.
-   *
-   * When a widget is moved, resized, deleted, appended, or altered, then this method is called
-   */
-  @Prop() onDashboardConfigurationChange?: (config: DashboardConfiguration) => void;
-
-  /**
-   * Whether the dashboard grid will stretch to fit.
-   *
-   * If stretch to fit is false, the dashboard grid will be the width in pixels.
-   * If not enough room is present, it will utilize scrollbars to allow access to the entire grid.
-   *
-   * If stretch to fit is true, the entire grid will scale proportionally to scale to the available space for the grid.
-   */
-  @Prop() stretchToFit: boolean = DEFAULT_STRETCH_TO_FIT;
-
-  /** Width of the dashboard, in pixels */
-  @Prop() width: number;
-
-  /** Width and height of the cell, in pixels */
-  @Prop() cellSize: number = DEFAULT_CELL_SIZE;
-
-  @Prop() move: (moveInput: MoveActionInput) => void;
-
-  /** List of ID's of the currently selected widgets. */
-  @State() selectedWidgetIds: string[] = [];
-
-  @State() currWidth: number;
   @Element() el!: HTMLElement;
 
-  /** The dashboard configurations current state. This is what the dashboard reflects as truth. */
-  @State() currDashboardConfiguration: DashboardConfiguration;
-  @State() intermediateDashboardConfiguration: DashboardConfiguration | undefined = undefined;
-
-  /** The currently active gesture */
-  @State() activeGesture: 'move' | 'resize' | 'selection' | undefined;
-
-  /**
-   * Selection gesture
-   */
-
-  @State() start: Position | undefined;
-  @State() end: Position | undefined;
-
-  /**
-   * Copy/paste action
-   */
-  // Currently selected group of widgets to copy
-  @State() copyGroup: Widget[] = [];
-  @State() numTimesCopyGroupHasBeenPasted = 0;
-
-  /**
-   * Move gesture
-   */
+  private undoQueue: UndoQueue = [];
```

Comment:
shouldn't the undo and redo queue be stored in the reducer rather than the dashboard component?